pkg/envoy/secretsync_test.go

```diff
@@ -0,0 +1,265 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package envoy
+
+import (
+	"context"
+	"io"
+	"testing"
+
+	cilium "github.com/cilium/proxy/go/cilium/api"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cilium/cilium/pkg/completion"
+	"github.com/cilium/cilium/pkg/envoy/xds"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	slim_metav1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/proxy/endpoint"
+)
+
+func Test_k8sToEnvoySecret(t *testing.T) {
+	envoySecret := k8sToEnvoySecret(&slim_corev1.Secret{
+		ObjectMeta: slim_metav1.ObjectMeta{
+			Name:      "dummy-secret",
+			Namespace: "dummy-namespace",
+		},
+		Data: map[string]slim_corev1.Bytes{
+			"tls.crt": []byte{1, 2, 3},
+			"tls.key": []byte{4, 5, 6},
+		},
+		Type: "kubernetes.io/tls",
+	})
+
+	require.Equal(t, "dummy-namespace/dummy-secret", envoySecret.Name)
+	require.Equal(t, []byte{1, 2, 3}, envoySecret.GetTlsCertificate().GetCertificateChain().GetInlineBytes())
+	require.Equal(t, []byte{4, 5, 6}, envoySecret.GetTlsCertificate().GetPrivateKey().GetInlineBytes())
+}
+
+func TestHandleSecretEvent(t *testing.T) {
+	logger := logrus.New()
+	logger.SetOutput(io.Discard)
+
+	xdsServer := &fakeXdsServer{}
+
+	syncer := newSecretSyncer(logger, xdsServer)
+
+	doneCalled := false
+	var doneError error
+
+	doneFunc := func(err error) {
+		doneCalled = true
+		doneError = err
+	}
+
+	// init
+	assert.Empty(t, syncer.secrets)
+
+	// upsert of unknown secret1
+	secret1 := testSecret("test1", "content")
+	err := syncer.handleSecretEvent(context.Background(), testEvent(secret1, resource.Upsert, doneFunc))
```

Comment:
Could we use table test for these?

Reply:
yep should be possible. i migrated it.

with the above mentions the set of tests changed a little bit (removing the ones that checked for updates (which no longer can be the case) and adding the ones for the discussed edge cases (nil checks etc))